bundler/lib/bundler/shared_helpers.rb

```diff
@@ -115,10 +115,6 @@ def filesystem_access(path, action = :write, &block)
       raise NoSpaceOnDeviceError.new(path, action)
     rescue Errno::ENOTSUP
       raise OperationNotSupportedError.new(path, action)
-    rescue Errno::EEXIST, Errno::ENOENT
-      raise
-    rescue SystemCallError => e
-      raise GenericSystemCallError.new(e, "There was an error #{[:create, :write].include?(action) ? "creating" : "accessing"} `#{path}`.")
```

Comment:
I'm a bit concerned about removing this fallback. Instead, what if we raised a new exception on `Errno::EROFS` that `write_lock` could rescue

Reply:
Yes, I was just thinking the same, sounds good 👍.